lib/ret_web/controllers/api/v1/bot_controller.ex

```diff
@@ -0,0 +1,143 @@
+defmodule RetWeb.Api.V1.BotController do
+  use RetWeb, :controller
+
+  alias RetWeb.Api.V1.{HubController}
+  alias Ret.{Hub}
+
+  @slack_api_base "https://slack.com"
+  @help_prefix "Hi! I'm the Hubs bot. I connect Slack channels with rooms on Hubs (https://hubs.mozilla.com/). Type `/hubs help` for more information."
+  @help_text "Command reference:\n\n" <>
+               "🦆 `/hubs` - Shows general information about the Hubs integration with the current Slack channel.\n" <>
+               "🦆 `/hubs help` - Shows this text you're reading right now.\n" <>
+               "🦆 `/hubs create` - Creates a default Hubs room and puts its URL into the channel topic. " <>
+               "Rooms created with `/hubs create` will inherit moderation permissions from this Slack channel and only allow Slack users in this channel to join the room.\n" <>
+               "🦆 `/hubs create [environment URL] [name]` - Creates a new room with the given environment and name, and puts its URL into the channel topic. " <>
+               "Valid environment URLs include GLTFs, GLBs, and Spoke scene pages.\n" <>
+               "🦆 `/hubs remove` - Removes the room URL from the topic and stops bridging this Slack channel with Hubs."
+
+  def create(conn, params) do
+    %{
+      # "/hubs"
+      "command" => command,
+      "channel_id" => channel_id,
+      # "help/create/etc. arg1 arg2 arg3"
+      "text" => text,
+      "team_id" => team_id
+    } = params
+
+    # TODO check if there's any double spaces
+    args = String.split(text, " ")
+    handle_command(List.first(args), channel_id, args, team_id)
+
+    conn
+    |> send_resp(200, "")
+  end
+
+  defp handle_command("help" = _command, channel_id, _args, _team_id) do
+    send_message_to_channel(channel_id, @help_text)
+  end
+
+  defp handle_command("" = _command, channel_id, _args, _team_id) do
+    send_message_to_channel(channel_id, @help_prefix)
+  end
+
+  defp handle_command("create" = _command, channel_id, args, team_id) do
+    %{
+      "channel" => %{
+        "name" => channel_name,
+        "topic" => %{"value" => topic}
+      }
+    } = get_channel_info(channel_id)
+
+    if has_hubs_url(topic) do
+      send_message_to_channel(
+        channel_id,
+        "A Hubs room is already bridged in the topic, so I am cowardly refusing to replace it."
+      )
+    else
+      # environmentURL = if (Enum.at(args, 1)) === nil, do "", else Enum.at(args, 1)
+      # name = if (Enum.at(args, 2)) === nil, do channelName, else Enum.at(args, 2)
+      params = %{
+        name: channel_name
+        # default_environment_gltf_bundle_url: sceneurl,
+        # scene_id: sceneid
+      }
+
+      {:ok, %{hub_sid: hub_sid} = new_hub} = HubController.create_new_room(params)
+      update_topic(channel_id, add_hub_topic(topic, Hub.url_for(new_hub)))
+
+      RetWeb.Api.V1.HubBindingController.bind_hub(%{
```

Comment:
Same here, move `bind_hub` into this controller, or into hub_binding.ex. In fact, it doesn't really make sense to create this struct to hold all the properties. You could just pass them directly to the function. That would also solve the awkward `"hub_id" => hub_sid` bit.